win32ss/user/user32/windows/cursoricon.c

```diff
@@ -1454,7 +1454,22 @@ CURSORICON_LoadImageW(
             RtlInitUnicodeString(&ustrRsrc, lpszName);
     }
 
-    if(hinst)
+    if(LDR_IS_RESOURCE(hinst))
+    {
+        /* We don't have a real module for GetModuleFileName, construct a fake name instead.
+        ** GetIconInfoEx reveals the name used by Windows.
+        */
+        LPCWSTR fakeNameFmt = sizeof(void*) > 4 ? L"\001%016IX" : L"\001%08IX";
+        ustrModule.MaximumLength = 42 * sizeof(WCHAR);
```

Comment:
In x64 it appears that only 18 characters (based on your format strings above, and including the terminating NULL char) are enough.